client.cpp

```diff
@@ -167,10 +168,14 @@ void Client::connectToBridgeTarget(FMQSockaddr_in6 addr)
            logger->logf(LOG_ERR, "Client connect error: %s", strerror(errno));
         return;
     }
+    assert(rc == 0);
 
-    if (rc == 0)
+    setBridgeConnected();
```

Comment:
I think `setBirdgeConnected()` needs to be after the setsocketopt and its error check?

Also, I'm assuming it works because it doesn't give an error, but can you set this option after connecting?

Reply:
I don't know, honestly, and could find out easily without testing it myself. Either way, the placement of  `setsock opt()` that you propose sounds much more sensible than what I did, so I changed the code accordingly.